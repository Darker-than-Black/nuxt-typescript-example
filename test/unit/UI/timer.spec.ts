import VueTestInstance from '~/test/factory/vue-instance'
import Timer from '~/components/UI/timer.vue'

describe('aside.vue', () => {
  const time: Date = new Date()
  const oneSecond: number = 1000
  let wrapper: any

  beforeEach(() => {
    wrapper = VueTestInstance(Timer, {
      options: {
        props: { time }
      }
    })
  })

  it('initialize correctly', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('check clean timer', () => {
    global.clearTimeout = jest.fn()
    wrapper.destroy()
    expect(global.clearTimeout).toHaveBeenCalledWith(wrapper.vm.timeoutFunc)
  })

  describe('test countdown()', () => {
    it('should recursive called countdown()', () => {
      // @ts-ignore
      global.setTimeout = jest.fn()
      wrapper.vm.countdown()
      expect(global.setTimeout).toHaveBeenCalledWith(wrapper.vm.countdown, oneSecond)
    })

    it('check work timer after deadline', () => {
      wrapper.setData({ deadline: time })
      wrapper.vm.countdown()
      expect(wrapper.vm.deadlineFail).toBe(true)
    })
  })
})
