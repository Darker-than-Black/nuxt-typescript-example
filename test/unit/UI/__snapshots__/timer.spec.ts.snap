// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aside.vue should match the snapshot 1`] = `
<div class="timer">
  <p class="timer__text">
    Час на обробку
  </p>
  <!---->
</div>
`;
